### PR TITLE
Added bucket stats event and viewer on debug page

### PR DESCRIPTION
**Pull request by lucaswoj**
screen shot 2015-09-25 at 12 13 27 pm

ref #1559 

cc @jfirebaugh @mourner

### CODE REVIEW

File: js/source/worker_tile.js [hunk 1]
```diff
@@ -226,7 +226,8 @@ WorkerTile.prototype.parse = function(data, layers, actor, callback) {
         callback(null, {
             elementGroups: elementGroups,
             buffers: buffers,
-            extent: extent
+            extent: extent,
+            bucketStats: self.bucketStats
```
Comment (lucaswoj) [inline js/source/worker_tile.js]: Is everyone ok with the perf impact of transferring this? Should be pretty small.

Comment (lucaswoj) [inline js/source/worker_tile.js]: We could add an explicit "enable" flag if necessary

**Comment by lucaswoj** [comment]
Are we ok with the perf impact of always sending bucket stats from the worker to the main thread? Or should we add an explicit "enable" flag?

**Comment by jfirebaugh** [comment]
It's not a large object right? Should be fine.

Surfacing this information is great. For a while I was kicking around the idea of writing a Chrome extension that added a "GL JS" tab to the dev tools that would surface this kind of information anywhere a mapbox-gl-js map was used. I even wrote a bit of code to prototype it: https://github.com/mapbox/mapbox-gl-js/compare/devtools

**Comment by lucaswoj** [comment]
> It's not a large object right? Should be fine.

Its one number per bucket. Looks like that's a pretty small amount of data on the absolute scale.

> For a while I was kicking around the idea of writing a Chrome extension that added a "GL JS" tab to the dev tools that would surface this kind of information anywhere a mapbox-gl-js map was used.

So cool! We should keep developing this if we have some spare cycles. Might be a good onboarding or interview sprint project.

**Comment by mourner** [comment]
Cool, :+1: I would add a slight semitransparent background to full label widths to make them more legible, and it's probably better to put the number in the beginning for quick scanning.

**Comment by lucaswoj** [comment]
@mourner How about some text outlines to match the basemap?

screen shot 2015-09-25 at 1 35 25 pm

**Comment by lucaswoj** [comment]
Just added the bucket stats benchmark script. Is that too much for this PR? Do we need to do some infrastructure work on our bench harness? :ship:?

**Comment by lucaswoj** [comment]
@jfirebaugh @mourner Is this good to :ship:? I'd like to pull it into `buffer2` to do some perf testing now that https://github.com/mapbox/mapbox-gl-js/pull/1568 has landed.

**Comment by jfirebaugh** [comment]
:+1:

**Comment by lucaswoj** [comment]
closed by aeecc12102a1415f4877a8cfb635da7c7a055693

**Pull request closed without merging**